tests/test_actions.py

```diff
@@ -0,0 +1,217 @@
+):
+    tracker.latest_message["intent"]["name"] = intent
+    tracker.latest_message["entities"].append(entity)
+    action = actions.ActionSetOnboarding()
+    actual_events = action.run(dispatcher, tracker, domain)
+    assert actual_events == expected_events
+
+
+def test_get_community_events(caplog):
+    """
+    Test that format of current events page is as expected and that parsing of a known events page finds the right events
+    """
+    with caplog.at_level(logging.WARNING):
+        community_events.get_community_events()
+    assert not caplog.text.startswith(
+        "Error when trying to parse event details from html."
+    )
+
+
+def test_parse_community_events(mocker):
+    mocker.patch.object(
+        community_events,
+        "get_community_page",
+        return_value=pickle.load(open("tests/data/events_page.pkl", "rb")),
+    )
+    actual_events = [
+        event.as_kwargs() for event in community_events.get_community_events()
+    ]
+
+    expected_events = [
+        community_events.CommunityEvent(
+            "WeAreDevelopers",
+            "Berlin",
+            "Germany",
+            "28 – June 29, 2021",
+            datetime.date.max,
+            "https://www.wearedevelopers.com/events/world-congress/",
+        ).as_kwargs()
+    ]
+
+    assert actual_events == expected_events
+
+
+def test_action_submit_subscribe_newsletter_form_notsubscribed(
+    tracker, dispatcher, domain, mailchimp_new_email
+):
+
+    tracker.slots["email"] = mailchimp_new_email
+    action = actions.ActionSubmitSubscribeNewsletterForm()
+    actual_events = action.run(dispatcher, tracker, domain)
+    assert actual_events == []
+    assert len(dispatcher.messages) == 1
+    assert dispatcher.messages[0]["template"] == "utter_confirmationemail"
+
+
+def test_action_submit_subscribe_newsletter_form_subscribed(
+    tracker, dispatcher, domain, mailchimp_subscribed_email
+):
+    tracker.slots["email"] = mailchimp_subscribed_email
+    action = actions.ActionSubmitSubscribeNewsletterForm()
+    actual_events = action.run(dispatcher, tracker, domain)
+    assert actual_events == []
+    assert len(dispatcher.messages) == 1
+    assert dispatcher.messages[0]["template"] == "utter_already_subscribed"
+
+
+def test_action_submit_sales_form(tracker, dispatcher, domain, mocker, gdrive_sheet):
+    company = "Sara CI"
+    use_case = "Unit Tests"
+    budget = "1000"
+    date = datetime.datetime.now().strftime("%d/%m/%Y")
+    # by using a random name we can verify that the line written to the spreadsheet is the one the test wrote
+    name = uuid.uuid1().hex
+    job_function = "Test Actions"
+    email = "example@rasa.com"
+
+    tracker.slots["budget"] = budget
+    tracker.slots["company"] = company
+    tracker.slots["business_email"] = email
+    tracker.slots["job_function"] = job_function
+    tracker.slots["person_name"] = name
+    tracker.slots["use_case"] = use_case
```

Comment:
this could be simplified with a 
```
collected_info = {
... dict of slots
}
tracker.slots.update(collected_info)
```
doesn't make it any shorter, but i think its the better pythonic way to do it